mapper/cumulocity/c8y_translator_lib/src/serializer.rs

```diff
@@ -0,0 +1,128 @@
+use chrono::prelude::*;
+use std::io::Write;
+use thin_edge_json::{json::ThinEdgeJsonError, measurement::GroupedMeasurementVisitor};
+
+pub struct C8yJsonSerializer {
+    buffer: Vec<u8>,
+    is_within_group: bool,
+    needs_separator: bool,
+    timestamp_present: bool,
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum C8yJsonSerializationError {
+    #[error(transparent)]
+    IoError(#[from] std::io::Error),
+
+    #[error(transparent)]
+    MeasurementCollectorError(#[from] MeasurementStreamError),
+
+    #[error(transparent)]
+    ThinEdgeJsonParseError(#[from] ThinEdgeJsonError),
+}
+#[derive(thiserror::Error, Debug, Eq, PartialEq)]
+pub enum MeasurementStreamError {
```

Comment:
Is that name actually right, what kind of stream do we have here?

Reply:
I assume `MeasurementCollectorError` should be accurate.